Tests/GoldEtlCsvTests.cpp

```diff
@@ -102,21 +102,42 @@ class Tests : public ::testing::Test, TestArgs {
                 if (testCsv.headerColumnIndex_[h] != SIZE_MAX && goldCsv.headerColumnIndex_[h] != SIZE_MAX) {
                     char const* a = testCsv.cols_[testCsv.headerColumnIndex_[h]];
                     char const* b = goldCsv.cols_[goldCsv.headerColumnIndex_[h]];
-                    if (_stricmp(a, b) != 0) {
-                        if (rowOk) {
-                            rowOk = false;
-                            printf("GOLD = %ls\n", goldCsv_.c_str());
-                            printf("TEST = %ls\n", testCsv_.c_str());
-                            AddTestFailure(__FILE__, __LINE__, "Difference on line: %zu", testCsv.line_);
-                            printf("    COLUMN                    TEST VALUE                            GOLD VALUE\n");
+                    if (_stricmp(a, b) == 0) {
+                        continue;
+                    }
+
+                    // Floating point may be inconsistently rounded by printf accross different platforms.
+                    // Do a rounding check by ensuring the difference between the two numebrs are less than 
+                    // the final digit +-1.001
+
+                    // Doubles should be good for 15 digits of precision.
+
+                    double testNumber = strtod(a, nullptr);
+                    double goldNumber = strtod(b, nullptr);
+                    const char* testDecimalAddr = strchr(a, '.');

```

Comment:
I think this fails for a==b=="0.000".  You could use sscanf() or something to test if the parse succeeded.